### PR TITLE
Add special Parameters modules support

### PR DESCRIPTION
**Pull request by younesbelkada**
This PR supports adding special `Parameter` modules when calling the context manager `init_empty_weights` that force assign the weights of each module into `nn.Parameter` module. For some specific usecases we might want to replce it with other modules such as `bnb.Int8Params` from `bitsandbytes`

Related PR: https://github.com/huggingface/transformers/pull/17901
Related discussion:  https://github.com/huggingface/transformers/pull/17901#discussion_r920230935

cc @sgugger @TimDettmers

### CODE REVIEW

File: src/accelerate/big_modeling.py [hunk 1]
```diff
@@ -39,9 +39,7 @@ def init_empty_weights(include_buffers: bool = False):
     Args:
         include_buffers (`bool`, *optional*, defaults to `False`):
             Whether or not to also put all buffers on the meta device while initializing.
-
     Example:
-
```
Comment (sgugger) [inline src/accelerate/big_modeling.py:44]: Leaves those poor lines alone :-)

**Review by sgugger — APPROVED** [review]
Good for me as it doesn't change anything for regular parameters! Thanks!

File: src/accelerate/big_modeling.py (hunk 1 above)
Comment (younesbelkada) [reply src/accelerate/big_modeling.py:44]: 🤣 yeah don't know what happened there

Comment (younesbelkada) [reply src/accelerate/big_modeling.py]: Fixed in aafea4a1e18ac6562bc678f3e607e0d2c5f89b38 ;)

**Pull request merged**